src/meds_torch/models/components/transformer_decoder.py

```diff
@@ -32,3 +298,286 @@ def forward(self, batch):
         batch[BACKBONE_TOKENS_KEY] = output
         batch[BACKBONE_EMBEDDINGS_KEY] = embeddings
         return batch
+
+    def generate(
+        self,
+        prompts: torch.Tensor,
+        mask: torch.Tensor | None = None,
+        get_next_token_time: Callable[[torch.Tensor, torch.Tensor], torch.Tensor] | None = None,
+        eos_token=None,
+        temperature=1.0,
+        filter_logits_fn: str | Callable = identity,
+        restrict_to_max_seq_len: bool = True,
+        filter_kwargs: dict = dict(),
+        cache_kv=True,
+        pad_value: int = 0,
+        **kwargs,
+    ) -> torch.Tensor:
+        """Generate tokens with either sequence length or time budget constraints.
```

Comment:
What's the point of this function rather than just using the base `generate`? That might be worth commenting on in this docstring.